### PR TITLE
Improve UI and explanations when alerts are disabled

### PR DESCRIPTION
**Pull request by digitalcora**
Currently a `none` communication mode is displayed the same as `email`, making it appear email delivery is enabled when in fact it isn't. There's no other indication in the UI that alerts are disabled, or explanation as to why, except when the reason is SMS opt-out and the user is still inside the opt-out freeze period.

With these changes, if a user's communication mode is `none`:

* their Settings page correctly has neither of the delivery options selected

* a flash appears on their Settings page explaining how this happened and what they can do to resolve it (for all circumstances, not just when inside the SMS freeze period)

* a flash appears on their Subscriptions page, which is the "home page" when signed in, explaining that alerts are disabled and to access the Settings page to resolve this

### CODE REVIEW

File: apps/concierge_site/lib/controllers/trip_controller.ex
Comment (arkadyan) [inline apps/concierge_site/lib/controllers/trip_controller.ex:29]: Would it be helpful if this included a link to your account settings?

**Review by arkadyan — APPROVED** [review]
Looks great to me 👍

File: apps/concierge_site/lib/controllers/trip_controller.ex
Comment (digitalcora) [reply apps/concierge_site/lib/controllers/trip_controller.ex:29]: Currently flash content can only be plain text (it is HTML-escaped on render), so before making that possible we'd have to go through and ensure all flash content is HTML-safe. I might look into that anyway since I am also waiting on a content review.

Comment (arkadyan) [reply apps/concierge_site/lib/controllers/trip_controller.ex:29]: Ah right, I forgot about that. Well I'll leave it up to you, but it doesn't seem that important.

**Pull request merged**
